### ISSUE TITLE
await makes code work inside a register-call, removing it brakes everything

### ISSUE THREAD

**Issue by Lilja** [fastify/help#799]
#### You have already researched for similar issues?
I'm sure this has happened before, but I don't feel empowered to find the right keywords to search this issue. Please forgive me.

#### What are you trying to achieve, or the steps to reproduce?

I'm encountering a bizzare error. I've the following "plugin"/scope:

```typescript
import fastifyWebsocket from "@fastify/websocket";
import fastifyPrintRoutes from "fastify-print-routes";

export const websocket2 = async (
  fastify: FastifyInstance,
  options: FastifyPluginOptions,
  done: (error?: FastifyError) => void
) => {
  fastify.register(import('@fastify/websocket'));
  await fastify.register(fastifyPrintRoutes, {
     useColors: false,
  });

  fastify.get("/sourceEvents", { websocket: true }, (connection, req) => {
    req.log.info("conn");
    connection.socket.on("message", (message) => {
      req.log.info({data: message.toString()}, "conn onMessage");
      connection.socket.send("hi from server");
    });
  });
  done()
}
```
Things are working as expected, however when I remove the await call to `fastifyPrintRoutes`, refresh my browser hitting the `/sourceEvents` websocket. Nothing is happening. I get the `conn`-log, but nothing is happening.

#### What was the result you received?
Nothing is happening.

#### What did you expect?
The websocket connection to run.

#### Context

* *node version*: 18.8.0
* *fastify version*: `4.8.1`
* `@fastify/websocket`: `7.1.0`
* `fastify-print-routes`: `2.0.4`
* *os*: Mac

This is the using the code from above, the `await fastify.register(fastifyPrintRoutes, {...})` commented, after waiting 10 seconds.
![image](https://user-images.githubusercontent.com/6134511/204907038-7b82d665-af3c-446b-9280-df43bd46ed37.png)

**Comment by metcoder95** [comment]
It is not bizarre but rather expected. On `v4` the `onRoute` hook became fully `sync`, meaning that all plugins that depend on the `onRoute` hook to execute their logic require the `Fastify#register` to await until the plugin is fully loaded before being able to use the plugin itself.

What is happening here is that, as you're awaiting for `printRoutes` plugin to be loaded, this causes to also the websocket plugin to be fully loaded (some sort of side-effect) when reaching the route that uses the websocket plugin.

In summary, if you're using a plugin that requires the use of the `onRoute` hook, you should always await for it to be fully loaded before start registering your routes

**Comment by Lilja** [comment]
@metcoder95 Thanks for your response!

I added `await register(import(...))`, yet I still am experiencing the aforementioned issue. Am I doing something else wrong?
